civitai-downloader/src/api/client.py

```
@@ -0,0 +1,619 @@
+            # Handle HTTP errors
+            if response.status_code == 401:
+                raise AuthenticationError("Invalid API key or authentication required")
+            elif response.status_code == 429:
+                raise RateLimitError("Rate limit exceeded")
+            elif response.status_code >= 400:
+                error_msg = f"API error {response.status_code}"
+                try:
+                    error_data = response.json()
+                    if "error" in error_data:
+                        error_msg = error_data["error"]
+                except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)